File: apollo/src/main/java/io/muun/apollo/domain/model/base/HoustonIdModel.java
package io.muun.apollo.domain.model.base;

import android.annotation.SuppressLint;
import android.support.annotation.Nullable;

import javax.validation.constraints.NotNull;

public class HoustonIdModel extends PersistentModel {

    @NotNull
    public Long hid;

    protected HoustonIdModel(@Nullable Long id, @NotNull Long hid) {
        super(id);
        this.hid = hid;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) {
            return true;
        }

        if (other == null || getClass() != other.getClass()) {
            return false;
        }

        return hid.equals(((HoustonIdModel) other).hid);
    }

    @Override
    public int hashCode() {

        return hid.hashCode();
    }

    @SuppressLint("DefaultLocale")
    @Override
    public String toString() {
        return String.format("<%s hid=%d>", getClass().getSimpleName(), hid != null ? hid : -1);
    }
}

File: apollo/src/main/java/io/muun/apollo/domain/model/base/HoustonIdModel.kt
package io.muun.apollo.domain.model.base

import android.annotation.SuppressLint

open class HoustonIdModel protected constructor(id: Long?, open val hid: Long) : PersistentModel(id) {

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }

        return if (other == null || javaClass != other.javaClass) {
            false
        } else hid == (other as HoustonIdModel).hid

    }

    override fun hashCode(): Int {

        return hid.hashCode()
    }

    @SuppressLint("DefaultLocale")
    override fun toString(): String {
        return String.format("<%s hid=%d>", javaClass.simpleName, hid)
    }
}
